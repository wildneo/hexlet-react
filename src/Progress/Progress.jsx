import React from 'react';

export default class Progress extends React.Component {
  render() {
    const { percentage } = this.props;
    const barStyle = { width: `${percentage}%` };

    return (
      <div className="progress">
        <div
          className="progress-bar"
          role="progressbar"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow={percentage}
          style={barStyle}
        />
      </div>
    );
  }
}
