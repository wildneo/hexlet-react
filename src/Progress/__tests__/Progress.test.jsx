import React from 'react';
import renderer from 'react-test-renderer';
import Progress from '../Progress';

test('Progress 1', () => {
  const component = renderer.create(<Progress percentage={25} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Progress 2', () => {
  const component = renderer.create(<Progress percentage={100} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
