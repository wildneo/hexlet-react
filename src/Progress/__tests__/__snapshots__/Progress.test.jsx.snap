// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress 1 1`] = `
<div
  className="progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={25}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "25%",
      }
    }
  />
</div>
`;

exports[`Progress 2 1`] = `
<div
  className="progress"
>
  <div
    aria-valuemax="100"
    aria-valuemin="0"
    aria-valuenow={100}
    className="progress-bar"
    role="progressbar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
</div>
`;
